### ISSUE TITLE
Test presence of conntrack module at insert time

### ISSUE THREAD

**Issue by cecio** [SYNwall/SYNwall#19]
I need to evaluate the possibility to test the presence of conntrack when UDP is enabled and used.

**Comment by camandel** [comment]
I modified the code to automatically load conntrack module only if parameter "enable_udp" is 1.
But even if the module is loaded correctly I always need (with newer kernel) add the iptable rule to get UDP traffic available.

**Comment by cecio** [comment]
yes....this needs to be investigated....

we may have several approach:
- implement the "make test" for UDP as well (ongoing with Issue#13)
- force an error at load time when UDP is enabled, but conntrack is not working (need to investigate how to check this out, may be trying to call a nf_* function
- force the module (but not only the module) load at startup

To tell the true, on some installation, even with new kernel, I don't have to do this manually. This is probably due to the presence of iptables/firewall services already using the conntrack.

Additional investigation is needed here...

**Comment by cecio** [comment]
I did some progress on this, but still investigating:

- `request_module` is not enough: the module is loaded, but it must be "registered" 
- the "registration" is probably done with `try_module_get,` but I'm still figuring out how to use it in the correct way...

**Comment by cecio** [comment]
ok, may be I found the way to do it...I'll implement it and we'll see

**Comment by cecio** [comment]
I did my homework on this, so to summarize:

- I was able to look for module and then use `try_module_get` to load modules
- this works fine, but unfortunately is not completely solving the issue: even if the module is loaded and marked as "used", the connection tracking does not start
- I did not found a supported way to start this tracking from inside the module, without adding additional dependencies (which I don't want to do)

At the end I decided to proceed in this way:
- if UDP is enabled, I'll add a check for the xt_conntrack module. If missing an error, suggesting to run the proper iptables command will be logged. Otherwise it will proceed, dropping additional errors as before if conntracking will fail.
